README.md

```diff
@@ -16,20 +16,276 @@ There are a number of other libraries out there to do this, but many of them are
 not well-tested or have some other deficiency. We needed a solid, quality event emitter
 mixin that we could use in a variety of projects.
 
-This event emitter will work both in node.js and in the browser.
+The event emitter mixin is a CommonJS module, and will work both in node.js and in the
+browser. For server-side usage, simply use the built-in `require` function to `require
+('silvermine-event-emitter')` and use the mixin. For browser environments, use a CommonJS
+module loader like [Browserify](http://browserify.org/) or
+[Webpack](https://webpack.github.io/).
 
-TODO: add comments about how to include it in your codebase (node - require, browser -
-browserify, other?). Also, add comments about needing a promise library / polyfill if you
-are going to use it in the browser.
+`silvermine-event-emitter` uses [native Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
+to make event listener executions asynchronous. Native Promises are available in node.js
+versions `0.12.18` and up, and in [most browsers](http://caniuse.com/#feat=promises)
+except Internet Explorer. If you need to support an environment that does not have native
+Promise support, you can easily add a [Promise polyfill](https://github.com/taylorhakes/promise-polyfill).
 
 ## How do I use it?
 
-Here's an example of how you can use this library:
+### Creating an event emitter
+
+The object returned by `require('silvermine-event-emitter')` is designed to be used as a
+mixin for javascript "classes". Here are a few different ways in which the mixin can be
+used:
+
+```js
+var EventEmitterMixin = require('silvermine-event-emitter'),
+    MyClass, myInstance;
+
+MyClass = function() {};
+
+// Copy functions from EventEmitterMixin to MyClass.prototype
+Object.assign(MyClass.prototype, EventEmitterMixin);
+// Or use underscore or lodash
+_.extend(MyClass.prototype, EventEmitterMixin);
+
+myInstance = new MyClass();
+myInstance.emit('hello');
+```
+
+Alternatively, if you prefer to use objects directly, without classes:
+
+```js
+var EventEmitterMixin = require('silvermine-event-emitter'),
+    myInstance;
+
+myInstance = Object.create(EventEmitterMixin, {
+   doSomething: function() {}
+});
+
+myInstance.emit('hello');
+```
+
+If you are using ES6, see
+[this guide on using mixins with ES6 classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Mix-ins).
+Here is an example of a mixin function that can be used to add event emitter methods to an
+ES6 class:
 
 ```js
-// TODO
+
+let EventEmitterMixin = require('silvermine-event-emitter');
+
+let mixinEventEmitter = (Base) => {
+   Object.assign(Base.prototype, EventEmitterMixin);
+   return Base;
+};
+
+class MyBaseClass {}
+
+class MyClass extends mixinEventEmitter(MyBaseClass) {
```

Comment:
Yeah, I wasn't sure about the capitalization since `mixinEventEmitter` is technically a function that returns a `class`. But you're right, the capital `M` is less jarring.